src/ikernel.ts

```diff
@@ -404,6 +417,17 @@ namespace IKernel {
      * The done handler for the kernel future.
      */
     onDone: () => void;
+
+    /**
+     * Register hook for IOPub messages.
+     * 
+     * @returns a disposable to unregister the hook.
+     * 
+     * #### Notes
+     * The most recently-registered hook is run first.
+     */
+    registerMessageHook(hook: (msg: KernelMessage.IIOPubMessage) => boolean): void;
```

Comment:
Good point about consistency. My hope was that most people would just use the kernel, so it was a concession to convenience for them, and it made the kernel more consistent (e.g., the comm registration returns a disposal). Okay, I'm figuring out your testing framework (looks really nice!), and then I'll start using it and make a choice between keeping the kernel as disposal returning or as register/remove functions. Thanks for your feedback!